src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -108,31 +108,22 @@ class SessionService(
     val hasSessionsWithCategoryGroups = sessionTemplates.any { it.permittedSessionCategoryGroups.isNotEmpty() }
     if (hasSessionsWithCategoryGroups) {
       return sessionTemplates.filter { sessionTemplate ->
-        sessionTemplate.permittedSessionCategoryGroups.isEmpty() || isPrisonerCategoryAllowedOnSession(sessionTemplate, prisonerCategory)
+        sessionValidator.isSessionAvailableToPrisonerCategory(prisonerCategory, sessionTemplate)
       }
     }
 
     return sessionTemplates
   }
 
-  fun isPrisonerCategoryAllowedOnSession(sessionTemplate: SessionTemplate, prisonerCategory: String?): Boolean {
```

Comment:
where has this gone?

Reply:
moved this into PrisonerSessionValidator